File: pkmnapi-api/src/requests/trainer_rewards.rs
use rocket_okapi::JsonSchema;
use serde::Deserialize;

use crate::requests::base::BaseRequest;

pub type TrainerRewardRequest =
    BaseRequest<TrainerRewardRequestType, TrainerRewardRequestAttributes>;

impl TrainerRewardRequest {
    pub fn get_reward(&self) -> u32 {
        self.data.attributes.reward
    }
}

#[derive(Debug, Deserialize, JsonSchema)]
#[allow(non_camel_case_types)]
pub enum TrainerRewardRequestType {
    trainer_rewards,
}

#[derive(Debug, Deserialize, JsonSchema)]
pub struct TrainerRewardRequestAttributes {
    pub reward: u32,
}

File: clients/js/src/requests/trainer_rewards.ts
import { BaseRequest } from './base';

export type TrainerRewardRequest = BaseRequest<TrainerRewardRequestType, TrainerRewardRequestAttributes>;

enum TrainerRewardRequestType {
    trainer_rewards = 'trainer_rewards',
}

export interface TrainerRewardRequestAttributes {
    reward: number;
}

export const TrainerRewardRequestDefault = (attributes: TrainerRewardRequestAttributes): TrainerRewardRequest => {
    return {
        data: {
            type: TrainerRewardRequestType.trainer_rewards,
            attributes,
        },
    };
};
